apps/src/templates/progress/ProgressTable.jsx

```diff
@@ -41,6 +43,12 @@ class ProgressTable extends React.Component {
     });
   }
 
+  // Make this more clever. I don't need to find all of them, I just need to know if there is at least ONE visible lesson.
+  checkVisibility = lessons => {
+    lessons.filter(lesson => this.props.lessonIsVisible(lesson, 'Student'))
+      .length;
```

Comment:
a few recommendations here:
1. make sure this returns a boolean to avoid surprises later
2. you can use `.some` instead of `.filter(...).length`
3. the best names for boolean (variables or methods) start with "is", "has", etc
4. you are not returning any value here!

So this could be:
```js
isAnyLessonVisible = lessons => (
  lessons.some(...)
)
```